### ISSUE TITLE
Redesign balanceDifference

### ISSUE THREAD

**Issue by frangio** [OpenZeppelin/openzeppelin-test-helpers#10]
`balanceDifference` currently has an uncomfortable API that uses a callback. I propose the following neater API.

- `balanceTracker(owner: Account): Promise<BalanceTracker>`
- `BalanceTracker.get(): Promise<BigNum>`

Example:
```javascript
const ownerBalance = await balanceTracker(owner);
expect(await ownerBalance.get()).to.bignumber.equal('0');
await send.ether(owner, '10');
expect(await ownerBalance.get()).to.bignumber.equal('10');
```

**Comment by frangio** [comment]
The idea is for `get()` to return the difference/delta with the moment the tracker was instantiated. It's not clear if a third `get()` returns the difference with the balance at instantiation or the previous `get()`. I'm guessing the latter semantics will be more useful. Perhaps we should name it `checkpoint()` (or the shorter `delta()`?) for that to be clearer.

```javascript
expect(await web3.eth.getBalance(owner)).to.bignumber.equal('1234');

const ownerBalance = await balanceTracker(owner);

expect(await ownerBalance.delta()).to.bignumber.equal('0');

await send.ether(owner, '10');

expect(await ownerBalance.delta()).to.bignumber.equal('10');

expect(await ownerBalance.delta()).to.bignumber.equal('0');

await send.ether(owner, '5');

expect(await ownerBalance.delta()).to.bignumber.equal('5');
```

**Comment by nventuro** [comment]
Closed via #13.
